src/main/java/input/ArgumentParser.java

```diff
@@ -0,0 +1,237 @@
+package input;
+
+import exceptions.InputParsingException;
+import options.ArrayOption;
+import options.BitsOption;
+import format.Format;
+import options.IOption;
+import options.IntOption;
+import utils.Flag;
+
+import java.util.Objects;
+
+/**
+ * Provides basic implementation of {@link IArgumentParser} interface.
+ */
+public class ArgumentParser implements IArgumentParser {
+
+    private Flag currentFlag;
+
+    private Format fromRepresentation;
+    private Format toRepresentation;
+    private IOption[] fromOptions;
+    private IOption[] toOptions;
+    private String inputFile;
+    private String outputFile;
+    private String delimiter;
+
+    private boolean shouldLookForFromOptions;
+    private boolean shouldLookForToOptions;
```

Comment:
**issue:** user expects to be able to use the options in arbitrary order

Having options tied to input and output format makes no sense. It would be better to parse all of the arguments first and then validate the resulting combination.